master/internal/api/socketlike.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/gorilla/websocket"
+)
+
+type ReadWriter interface {
+	ReadJSON(interface{}) error
+	Write(interface{}) error
+}
+
+type WrappedWebsocket struct {
+	*websocket.Conn
+}
+
+func (w *WrappedWebsocket) Write(msg interface{}) error {
+	pm, ok := msg.(*websocket.PreparedMessage)
```

Comment:
This looks right, but the location of the code is a bit off.

Anytime I see C code that is typecasting `void*`'s, or go code that is typecasting `interface{}`'s, I expect the code which makes that typecasting to also live nearby.

Basically, somewhere there's a prepare function that should always be used with WrappedWebsocket that makes sure that the msg passed to Write is actually a PreparedMessage.  Why is that code not right next to this code?

Or in this case, why is this code not right next to that code?  It looks like this interface is only used in `internal/stream` so probably it should live in that directory, not here.